File: Backend/models/GameSession.js
const mongoose = require('mongoose');

const gameSessionSchema = new mongoose.Schema({
  sessionId: {
    type: String,
    required: true,
    unique: true
  },
  players: [{
    name: {
      type: String,
      required: true
    },
    score: {
      type: Number,
      default: 0
    },
    answers: [{
      questionId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Question'
      },
      userAnswer: String,
      isCorrect: Boolean,
      answeredAt: {
        type: Date,
        default: Date.now
      }
    }]
  }],
  questions: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Question'
  }],
  currentQuestionIndex: {
    type: Number,
    default: 0
  },
  currentPlayerIndex: {
    type: Number,
    default: 0
  },
  status: {
    type: String,
    enum: ['waiting', 'active', 'completed'],
    default: 'waiting'
  },
  totalQuestions: {
    type: Number,
    default: 15
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  completedAt: {
    type: Date
  }
}, {
  timestamps: true
});

// Add index for better performance
gameSessionSchema.index({ sessionId: 1 });
gameSessionSchema.index({ status: 1 });
gameSessionSchema.index({ createdAt: 1 }, { expireAfterSeconds: 86400 }); // Auto-delete after 24 hours

module.exports = mongoose.model('GameSession', gameSessionSchema);

File: trivia_showdown/lib/models/game_session.dart
class GameSession {
  final String sessionId;
  final List<Player> players;
  final int currentQuestionIndex;
  final int totalQuestions;
  final String status;
  final String currentPlayer;

  GameSession({
    required this.sessionId,
    required this.players,
    required this.currentQuestionIndex,
    required this.totalQuestions,
    required this.status,
    required this.currentPlayer,
  });

  factory GameSession.fromJson(Map<String, dynamic> json) {
    return GameSession(
      sessionId: json['sessionId'],
      players: (json['players'] as List)
          .map((player) => Player.fromJson(player))
          .toList(),
      currentQuestionIndex: json['currentQuestionIndex'],
      totalQuestions: json['totalQuestions'],
      status: json['status'],
      currentPlayer: json['currentPlayer'],
    );
  }
}

class Player {
  final String name;
  final int score;
  final List<dynamic> answers;

  Player({
    required this.name,
    required this.score,
    required this.answers,
  });

  factory Player.fromJson(Map<String, dynamic> json) {
    return Player(
      name: json['name'],
      score: json['score'],
      answers: json['answers'],
    );
  }
}